tests/test_validate_parameters.py

```diff
@@ -10,14 +10,53 @@
 
 import pytest
 import yaml
+from click.testing import CliRunner
 
+from reana_client.utils import cwl_load
 from reana_client.validation import (
     _validate_dangerous_operations,
     _validate_serial_parameters,
     _validate_yadage_parameters,
+    _validate_cwl_parameters,
 )
 
 
+def test_validate_parameters_cwl(
+    create_cwl_yaml_workflow_schema,
+    cwl_workflow_spec_correct_input_param,
+    cwl_workflow_spec_wrong_input_param,
+    capsys,
+):
+    """Validate parameters for CWL workflows."""
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        with open("test.cwl", "w") as f:
+            f.write(cwl_workflow_spec_correct_input_param)
+        reana_yaml = yaml.load(create_cwl_yaml_workflow_schema, Loader=yaml.FullLoader)
+        reana_yaml["workflow"]["specification"] = cwl_load(
+            reana_yaml["workflow"].get("file")
+        )
+        _validate_cwl_parameters(reana_yaml)
+        captured = capsys.readouterr()
+        assert not captured.out
+
+    # Wrong Input parameter used
+    with runner.isolated_filesystem():
+        with open("test.cwl", "w") as f:
+            f.write(cwl_workflow_spec_wrong_input_param)
+        reana_yaml = yaml.load(create_cwl_yaml_workflow_schema, Loader=yaml.FullLoader)
+        reana_yaml["workflow"]["specification"] = cwl_load(
+            reana_yaml["workflow"].get("file")
+        )
+        with pytest.raises(SystemExit) as error:
+            _validate_cwl_parameters(reana_yaml)
+        assert error.value.code == 1
+
+    # Wrong file path used
+    reana_yaml = yaml.load(create_cwl_yaml_workflow_schema, Loader=yaml.FullLoader)
+    pytest.raises(SystemExit, _validate_cwl_parameters, reana_yaml)
```

Comment:
Could you check the output and verify that is what we expect? We do the same for other tests.

Reply:
The error message is not captured in `capsys`. I am still looking into that. Only the error code is there.

Reply:
@mvidalgarcia  I am just getting the error code that is `1` here not the error message that I get when I run  `reana-client validate --environments`.

Reply:
@mvidalgarcia The reason why I am not getting the error message is that when `run_command` method runs the `cwltool` command, it throws an error and in case of error we are just returning the `err.returncode` from there.

I can do the `subprocess` call direct from `validate_cwl` instead of calling `run_command`. Wdyt?

Reply:
Maybe you can enrich `run_command` with a kwarg to return also the output and use it?